portfolio/src/main/webapp/script.js

```diff
@@ -34,3 +34,62 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact; 
 }
+
+const size = 30;
+var canvas;
+var ctx;
```

Comment:
Better to make them local variables and get canvas and context each time in different function